Enhance publication lists to support showing abstracts, whenever available

igorburago opened the pull request:
Within the interface of a publications section, allow the user to view
the text of an entry’s abstract, if present, in the same fashion it is
already possible to view the entry’s BibTeX snippet.

Namely, add a button in the row of trailing “extra” links at the end
of each entry, pressing which toggles the visibility of the abstract.
The block containing the text of the abstract is placed right under
the entry itself (before the BibTeX snippet), and is hidden by default.
Both the abstract and the BibTeX blocks can be shown or hidden
independently from each other.

Commit:
Present abstracts for publication entries whenever available

Since there may now be two kinds of supplemental content to
a bibliography entry (the abstract text and the BibTeX snippet),
both of them are joined into the appendix block following the
entry, with the display of each appendix item being toggled
independently by the user via the buttons in the row of “extra”
links provided next to bibliographical data.

Commit:
Extend JS support for toggling display of abstracts in bibliographies

Generalize the toggleBibtexDisplay() function into toggleAppendixDisplay(),
making both the old toggleBibtexDisplay() function and the new
toggleAbstractDisplay() function special cases of toggleAppendixDisplay().

Commit:
Ensure the child selector is used for bibliography extras in CSS

iburago commented:
@dcallega: Although this changeset can already be merged in as is,
I still consider it as work-in-progress. For instance, I am going to
improve the handling of multi-paragraph abstracts (which currently
get merged into a single paragraph).

If you have time to try it out, let me know what you think of it, both
in terms of functionality and style.

Commit:
scripts/bibtex2yaml: Keep empty lines between paragraphs in detexify()

There is a slight potential disadvantage to this change. We do not
separate the LaTeX-formatted BibTeX fields into those that are
essentially one-line only (e.g., ‘title’) and those that may contain
a piece of text, generally, of multiple paragraphs (e.g., ‘abstract’).
This means that if a blank line is introduced into the value of
a naturally one-line field, it will be preserved. Given that mistakes
like that should be easily catchable, though, maintaining which field
is which in the script does not seem to be worth it at the moment.

Commit:
Support multi-paragraph abstracts in bibliography entries

The content of an entry’s ‘abstract’ field in bibliography data is assumed
to be the result of the conversion from LaTeX source to plain text, with
paragraphs (when there is more than one) separated by empty lines
(i.e., two consecutive newline characters).

Commit:
Use 'div' tags for BibTeX snippets in bibliographies

At this point, we do not inherit anything important from the default
styling of the previously used ‘pre’ tag, except for the monospace
font. Hence, specify that explicitly and switch over to ‘div’ to improve
consistency between appendix items in bibliography entries.

Commit:
Equalize padding of different appendix items in bibliography entries

iburago commented:
@dcallega: All right, I am done with all of the improvements I had in
mind for this changeset. Once you get a chance to review it, feel free
to merge it in, if it looks good to you.

iburago commented:
@dcallega: All right, I am done with all of the improvements I had in
mind for this changeset. Once you get a chance to review it, feel free
to merge it in, if it looks good to you.

iburago commented:
@All right, I am done with all of the improvements I had in mind for
this changeset. Once you get a chance to review it, feel free to merge
it in if it looks good to you.

dcallega commented:
I tried running make bib script and then the hugo server. This is the message I get. Am I missing some package? I am sorry, I don't have much time due to coming deadline, so I didn't try debugging.

Do you have a VM to try this in? I am using Ubuntu 18.04.

```
Building sites … ERROR 2020/03/11 10:46:53 Failed to add template "partials/bib/getfullname.html" in path "/home/ias/git/website/layouts/partials/bib/getfullname.html": template: partials/bib/getfullname.html:3: unexpected "=" in operand
ERROR 2020/03/11 10:46:53 Failed to add template "partials/bib/getpeoplepages.html" in path "/home/ias/git/website/layouts/partials/bib/getpeoplepages.html": template: partials/bib/getpeoplepages.html:4: unexpected "=" in operand
ERROR 2020/03/11 10:46:53 Failed to add template "partials/bib/names.html" in path "/home/ias/git/website/layouts/partials/bib/names.html": template: partials/bib/names.html:33: unexpected "=" in operand
ERROR 2020/03/11 10:46:53 Failed to add template "section/people.html" in path "/home/ias/git/website/layouts/section/people.html": template: section/people.html:13: unexpected "=" in operand
ERROR 2020/03/11 10:46:53 partials/bib/getfullname.html : template: partials/bib/getfullname.html:3: unexpected "=" in operand
ERROR 2020/03/11 10:46:53 partials/bib/getpeoplepages.html : template: partials/bib/getpeoplepages.html:4: unexpected "=" in operand
ERROR 2020/03/11 10:46:53 partials/bib/names.html : template: partials/bib/names.html:33: unexpected "=" in operand
ERROR 2020/03/11 10:46:53 section/people.html : template: section/people.html:13: unexpected "=" in operand
ERROR 2020/03/11 10:46:53 Error while rendering "section" in "people/": template: "section/people.html" is an incomplete or empty template
ERROR 2020/03/11 10:46:53 Error while rendering "page" in "people/": template: people/single.html:96:9: executing "people/single.html" at <partial "bib/section...>: error calling partial: template: partials/bib/section.html:1:20: executing "partials/bib/section.html" at <partial "bib/getpeop...>: error calling partial: template: "partials/bib/getpeoplepages.html" is an incomplete or empty template
ERROR 2020/03/11 10:46:53 Error while rendering "page" in "people/": template: people/single.html:17:35: executing "people/single.html" at <partial "people/year...>: error calling partial: template: partials/people/yearrange.html:7:39: executing "partials/people/yearrange.html" at <$t0.Unix>: can't evaluate field Unix in type *errors.errorString
ERROR 2020/03/11 10:46:53 Error while rendering "section" in "publications/": template: section/publications.html:8:9: executing "section/publications.html" at <partial "bib/section...>: error calling partial: template: partials/bib/section.html:1:20: executing "partials/bib/section.html" at <partial "bib/getpeop...>: error calling partial: template: "partials/bib/getpeoplepages.html" is an incomplete or empty template
Total in 160 ms
Error: Error building site: logged 8 error(s)
```

igorburago commented:
No special setup is necessary for building the site from the sources,
except for a recent version of Hugo. In the error messages, it complains
of variable overwrites in templates, which are supported by Hugo since
[version 0.48][1]. (It is a [Go 1.11 feature of Go templates][2], and 0.48 is the first
Hugo release that was built with Go 1.11 or newer.)

[1]: https://gohugo.io/news/0.48-relnotes/
[2]: https://golang.org/doc/go1.11#text/template

If you are using a Hugo binary installed from a Linux distribution package,
it might be outdated. First of all, check its version by running

	$ hugo version

If it is indeed older than 0.48, you will need to install a recent version
of Hugo. Luckily, it is easy to do, as Hugo does not require installation,
really: all you need is to have a copy of its binary available somewhere.

You can easily obtain it by using our own [`gethugo` script][3]. Just run it
from any directory you have the rights to write in, and it will download
the binary from the [latest release][4] of Hugo to that directory. After that,
just use this new binary as usual (instead of that old one in `$PATH`).

[3]: https://github.com/uci-iasl/website/blob/release/scripts/gethugo
[4]: https://github.com/gohugoio/hugo/releases/latest

igorburago commented:
@dcallega: Have you had any luck building with a recent version of Hugo?

igorburago commented:
All right, I’m merging this pull request to unblock other changesets
ready to be merged in, that depend on this one. If you would like to
change how abstracts are displayed, feel free to open an issue or
another pull request.
